src/main/java/fridayproject/Friday.java

```diff
@@ -0,0 +1,200 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+/**
+ * Represents the main class of the program.
+ */
+public class Friday {
+    private static final String FILE_PATH = "data/friday.txt";
+
+    /**
+     * Saves the tasks to a file.
+     * @param tasks
+     * @throws IOException
+     */
+    public static void saveTasksToFile(ArrayList<Tasks> tasks) throws IOException {
+        FileWriter fileWriter = new FileWriter(FILE_PATH);
+        for (Tasks task : tasks) {
+            fileWriter.write(task.toFileString() + "\n");
+        }
+        fileWriter.close();
+    }
+
+    /**
+     * Loads the tasks from a file.
+     * @return
+     * @throws IOException
+     */
+    public static ArrayList<Tasks> loadTasksFromFile() throws IOException {
+        File file = new File(FILE_PATH);
+        ArrayList<Tasks> tasks = new ArrayList<>();
+        if (!file.exists()) {
+            file.getParentFile().mkdirs();
+            file.createNewFile();
+            return tasks;
+        }
+
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNextLine()) {
+            String taskString = scanner.nextLine();
+            String[] parts = taskString.split(" \\| ");
+            try {
+                switch (parts[0]) {
+                case "T":
+                    if (parts.length >= 3) {
+                        Tasks todo = new Todo(parts[2]);
+                        if(parts[1].equals("1")) {
+                            todo.markAsDone();
+                        }
+                        tasks.add(todo);
+                    } else {
+                        System.out.println("Error loading task: " + taskString);
+                    }
+                    break;
+                case "D":
+                    if (parts.length >= 4) {
+                        Tasks deadline = new Deadline(parts[2], parts[3]);
+                        if (parts[1].equals("1")) {
+                            deadline.markAsDone();
+                        }
+                        tasks.add(deadline);
+                    } else {
+                        System.out.println("Error loading task: " + taskString);    
+                    }
+                    break;
+                case "E":
+                    if (parts.length >= 5) {
+                        Tasks event = new Event(parts[2], parts[3], parts[4]);
+                        if (parts[1].equals("1")) {
+                            event.markAsDone();
+                        }
+                        tasks.add(event);
+                    } else {
+                        System.out.println("Error loading task: " + taskString);
+                    }
+                    break;
+                default:
+                    System.out.println("Error loading task: " + taskString);
+                    break;
+                }
+            } catch (Exception e) {
+                System.out.println("Error loading task: " + taskString);
+            }
```

Comment:
this looks like something that can be abstracted further, in the name of OOP and modularity